File: backend/entity/SubDistrict.go
package entity

import "gorm.io/gorm"

type SubDistrict struct {
	gorm.Model
	NameTH     string   `json:"name_th"`
	NameEN     string   `json:"name_en"`
	DistrictID uint     `json:"district_id"`
	District   District `gorm:"foreignKey:DistrictID"`
	PostcodeID uint     `json:"postcode_id"`
	Postcode   Postcode `gorm:"foreignKey:PostcodeID"`
}

File: frontend/src/interfaces/SubDistrict.ts
import type { DistrictInterface } from "./District";
import type { PostcodeInterface } from "./Postcode";

export interface SubDistrictInterface {
    ID: number;
    name_th: string;
    name_en: string;
    district_id: number;
    District?: DistrictInterface;
    postcode_id: number;
    Postcode?: PostcodeInterface;
}